### DISCUSSION TITLE
Help with contributing (translation)?

### DISCUSSION THREAD

**Discussion by DevWannabe-dot** [qsniyg/maxurl#892] [Q&A] [answered]
Hey! Good afternoon everyone! 

I've been working on en -> pt_BR translation, because it's my mother tongue and I dominate it pretty well, so I thought I could help, but, something seems off?
![1-comment or source text, 2-id, 3-target text](https://user-images.githubusercontent.com/64389041/132028842-7af9ebfd-1390-42c2-9eb8-a7bfb8eae52c.png)
If line 19 is a comment or source text, line 20 is the id for a certain text to be translated and 21 is the entry for the target (translated) text, why, further in the code, does it look like this?
![Sorted 2-1-3 of what it was on the first image](https://user-images.githubusercontent.com/64389041/132029161-2e8b7cf6-7f47-4c6a-bd20-2f0d7e98345b.png)

Also I'm aware of the pull request for chinese translation, #861, that has not yet been merged and that the way to contribute with a translation would be changing, so, it'd be a waste of time now, right? Thanks in advance.

**Comment by qsniyg** [comment, marked as answer]
Thank you very much for offering to translate!!

Sorry, yeah I realize this is not very consistent. At one point I thought it'd be a good idea to use message IDs: https://github.com/qsniyg/maxurl/issues/90

But as mentioned in the issue, this is not a good idea. I have to move them over to the second way (msgid = english string, rather than ID). I can do that on my end though, so (hopefully) no need to worry about that :)

>so, it'd be a waste of time now, right?

Actually no! I'm holding off on adding new features until the TS rewrite is complete, so other than possibly taking a while to merge, the translation strings won't change in any significant way until I can merge the translations :)

**Comment by Korb** [comment]
@qsniyg, I read the [Translations](https://github.com/qsniyg/maxurl/blob/master/CONTRIBUTING.md#translations) section in `CONTRIBUTING.md` about your practice of translating via `.po`-files. It's the simplest solution imaginable, but unfortunately, it's also the most inconvenient in the long run. This method doesn't allow for many things:

1. Ability to use a unified glossary (unified translation rules for specific words and expressions across the entire project, including the UI, documentation, your website, and userscript);
2. Track changes to the source translation, marking the corresponding translations as outdated;
3. Automatically track the translation progress of incomplete (or no longer up-to-date) translations;
4. Translators cannot receive notifications about new segments requiring translation;
5. No ability to discuss specific strings or rules affecting the project as a whole;
6. No ability to search for other segments by keywords, including in other languages.

Please consider integrating one of the free _(for FOSS projects)_ translation platforms:

1. [Weblate](https://weblate.org) (used by F-Droid too, open-source, self-hostable),
2. [Crowdin](https://crowdin.com/page/open-source-project-setup-request) or
3. [Transifex](https://www.transifex.com/open-source/) (both proprietary but free for FLOSS projects),
4. Lokalise (“[We offer Lokalise for free to open source projects](https://lokalise.com/pricing/#radix-_r_5i_)”)

Don't forget about Instructions for translators (an example of [a good instruction](https://hosted.weblate.org/projects/breezy-weather/#information)).

---
_Tags: L10n, i18n
Related do #654, #860 and #1515._

**Reply by qsniyg** [reply]
Thank you! I'll take a look at these. I agree, the translation workflow is far from ideal at the moment.
